java/AvroOneTopicManySubjects/src/main/java/com/amazonaws/services/msf/StreamingJob.java

```diff
@@ -0,0 +1,352 @@
+     * @param topic               topic name
+     * @param schemaRegistryUrl   Confluent Schema Registry region
+     * @param kafkaProducerConfig configuration passed to the Kafka Producer
+     * @return a KafkaSink
+     */
+    private static <T extends SpecificRecord> KafkaSink<T> keyedKafkaSink(String bootstrapServers, String topic, String schemaRegistryUrl, Properties kafkaProducerConfig, Map<String, Object> schemaRegistryConfig) {
+        RecordNameSerializer<T> serializer = new RecordNameSerializer<T>(schemaRegistryUrl, topic, schemaRegistryConfig);
+        return KafkaSink.<T>builder().setBootstrapServers(bootstrapServers).setRecordSerializer(serializer).setKafkaProducerConfig(kafkaProducerConfig).build();
+    }
+
+    private static float randomSensorReading(float min, float max) {
+        float n = new Random().nextFloat();
+        return min + n * (max - min);
+    }
+
+    private static DataGeneratorSource<RoomTemperature> temperatureSamples() {
+        GeneratorFunction<Long, RoomTemperature> generatorFunction = index -> new RoomTemperature(new Random().nextInt(NUM_ROOMS), "room-1", randomSensorReading(17, 40), Instant.now());
+        return new DataGeneratorSource<RoomTemperature>(generatorFunction, Integer.MAX_VALUE, RateLimiterStrategy.perSecond(1), TypeInformation.of(RoomTemperature.class));
+    }
+
+    private static DataGeneratorSource<AirQuality> airQualitySamples() {
+        GeneratorFunction<Long, AirQuality> generatorFunction = index -> new AirQuality(new Random().nextInt(NUM_ROOMS), "room-1", randomSensorReading(1, 10), Instant.now());
+        return new DataGeneratorSource<AirQuality>(generatorFunction, Integer.MAX_VALUE, RateLimiterStrategy.perSecond(1), TypeInformation.of(AirQuality.class));
+    }
+
+    private static void ensureTopicExist(String bootstrapServers, String topicName, int partitions, short replicationFactor) throws Exception {
+        Properties config = new Properties();
+        config.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+
+        try (AdminClient adminClient = AdminClient.create(config)) {
+            // Create the topic if it does not exist
+            NewTopic newTopic = new NewTopic(topicName, partitions, replicationFactor);
+            adminClient.createTopics(Collections.singleton(newTopic)).all().get();
+            System.out.println("Topic '" + topicName + "' created successfully.");
+        } catch (ExecutionException e) {
+            if (!(e.getCause() instanceof TopicExistsException)) {
+                throw e;
+            }
+            // This is fine because we want the topic to exist
+        }
+    }
+}
+
+/**
+ * Option type is a generic container to hold deserialized records.
+ * It can be useful to introduce fields into this type to handle partitioning
+ * strategies and event time extraction. However, for those scenarios to work
+ * all subjects should have a standard set of fields.
+ */
+class Option {
```

Comment:
Maybe you can use `org.apache.flink.types.SerializableOptional<T>` that comes with Flink

Reply:
`Option` type in this PR is a container type to hold *any* possible deserialized value. `SerializableOptional<T>` is for optional values. I guess it would not be the right choice here? Am I missing something? 👀 🙏🏾 

BTW: I could have used `Object` instead of creating an `Option` with the `Object` type `value` field. However, having `Option` type helps if we want to generate watermarks via source operator using a common timestamp field. 

Is there a better way to do this?